src/frequenz/sdk/data_ingestion/resampling/subscriptions.py

```diff
@@ -0,0 +1,40 @@
+"""
+Common model definitions used for resampling purposes.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+from frequenz.channels import Receiver, Sender
+
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """Subscription for receiving data for a specific time series id."""
+
+    time_series_id: TimeSeriesId
+    start_time: Optional[datetime] = None
+
+
+@dataclass(frozen=True)
+class SubscriptionResponse:
+    """Response to a subscription request with a receiver for ingesting data."""
+
+    receiver: Receiver[Optional[Sample]]
+    subscription: Subscription
+
+
+@dataclass(frozen=True)
+class SubscriptionRequest:
+    """Request specifying the object of subscription and where to send the response."""
+
+    sender: Sender[SubscriptionResponse]
```

Comment:
A request definitely can't pass a `Sender`. Even if this might be supported now, when we go distributed, we'll probably have to do it differently. So we will very likely pass channels names in here, see the other comment. But still to be confirmed.